pkg/tenantctl/controller.go

```diff
@@ -184,21 +184,21 @@ func (c *Controller) syncHandler(key string) error {
 
 	tenant := origTenant.DeepCopy()
 
-	c.logPrintln(tenant, "started reconciliation")
-	defer c.logPrintln(&api.Tenant{ObjectMeta: *tenant.ObjectMeta.DeepCopy()}, "finished reconciliation")
+	klog.V(4).Infof(c.formatLog(tenant, "started reconciliation"))
+	defer klog.V(4).Infof(c.formatLog(&api.Tenant{ObjectMeta: *tenant.ObjectMeta.DeepCopy()}, "finished reconciliation"))
 
 	// the configuration should be loaded once per sync to avoid inconsistencies
 	// in case of concurrent configuration changes
 	config, err := c.getClientConfig(c.factory, tenant.GetNamespace())
 	if err != nil {
-		c.logPrintln(tenant, err)
+		klog.V(2).ErrorS(err, c.formatLog(tenant))
```

Comment:
Does it make sense to make this dependent on the verbosity level? I think it should always be logged.

Applies to other places, too.